runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1131,13 +1131,18 @@ MM_ObjectAccessBarrier::indexableStoreI64(J9VMThread *vmThread, J9IndexableObjec
 void
 MM_ObjectAccessBarrier::copyObjectFieldsToFlattenedArrayElement(J9VMThread *vmThread, J9ArrayClass *arrayClazz, j9object_t srcObject, J9IndexableObject *arrayRef, I_32 index)
 {
-	UDATA const objectHeaderSize = J9VMTHREAD_OBJECT_HEADER_SIZE(vmThread);
+	UDATA elementStartOffset = J9VMTHREAD_OBJECT_HEADER_SIZE(vmThread);
 	U_8 *elementAddress = (U_8*)indexableEffectiveAddress(vmThread, arrayRef, index, J9ARRAYCLASS_GET_STRIDE((J9Class *) arrayClazz));
 	IDATA elementOffset = (elementAddress - (U_8*)arrayRef);
 	J9Class *elementClazz = J9GC_J9OBJECT_CLAZZ_THREAD(srcObject, vmThread);
 	Assert_MM_true(J9_IS_J9CLASS_VALUETYPE(elementClazz));
 	Assert_MM_true(elementClazz == arrayClazz->leafComponentType);
-	copyObjectFields(vmThread, elementClazz, srcObject, objectHeaderSize, (j9object_t) arrayRef, elementOffset);
+
+	if (J9CLASS_HAS_4BYTE_PREPADDING(elementClazz)) {
```

Comment:
Does this mean that arrays of padded VTs will pad the start of each element?

Reply:
No VT array elements will not include pre-padding. See https://github.com/eclipse/openj9/blob/master/runtime/vm/createramclass.cpp#L3039

Reply:
I suppose since VTs are immuteable, there are no volatile/atomic concerns.